x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/session/VersionCompatibilityChecks.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.session;
+
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.sql.proto.SqlVersion;
+
+/**
+ * The class contains checks gating newly introduced functionality that is exposed to clients, but not supported by those older than the
+ * version introducing the respective feature.
+ * Each new feature will have a method added, taking a version and returning a boolean reflecting that version's support of the feature.
+ * The comparison can only have the equal added once the feature is backported (see JdbcTestUtils.java). Example:
+ * <code>
+ *     public static boolean supportsUnsignedLong(SqlVersion version) {
+ *         // TODO: add equality only once actually ported to 7.11
+ *         return VERSION_INTRODUCING_UNSIGNED_LONG.compareTo(version) &lt; 0;
+ *     }
+ * </code>
+ */
+public final class VersionCompatibilityChecks {
+
+    private VersionCompatibilityChecks() {}
+
+    /**
+     * Is the provided {@code dataType} being supported in the provided {@code version}?
+     */
+    public static boolean isTypeSupportedInVersion(DataType dataType, SqlVersion version) {
```

Comment:
This is a superfluous class that I adopted from the unsigned_long PR, and is not currently useful, as DATETIME_NANOS are available even for older drivers, you just get only the milliseconds. I will remove it, doesn't make sense to add something now for potential future use.

Reply:
This is a superfluous class that I adopted from the unsigned_long PR, and is not currently useful, as DATETIME_NANOS are available even for older drivers, you just get only the milliseconds. I think I will remove it, along with all the other changes in Analyzer, to me it doesn't make sense to add something now for potential future use. @bogdan @costin what do you think?

Reply:
> as DATETIME_NANOS are available even for older drivers

Just to put it out there: this might justify reverting that UL PR stub chunks, if it still makes sense. The PR would only be useful for the bwc handling of new types.